### PR TITLE
Make graph xscale and yscale settable

### PR DESCRIPTION
**Pull request by dqnykamp**
## Summary

Makes the `xscale` and `yscale` properties of a `<graph>` settable. Previously they were read-only derived values (`xMax − xMin` and `yMax − yMin`) with no inverse definition, so binding to them or targeting them with `<updateValue>` did nothing.

## Behavior

When `xscale` is set to `s`, the midpoint of the limits `(xMin + xMax) / 2` is held fixed and both ends move symmetrically so that `xMax − xMin === s`:

- `newXMin = mid − s/2`
- `newXMax = mid + s/2`

`yscale` behaves analogously with `yMin`/`yMax`.

The inverse writes desired values back to the existing `xMin`/`xMax` (and `yMin`/`yMax`) dependencies, so the established inverse logic—including the `fixAxes` refusal and graph-parent handling—is reused. Non-finite requested values, midpoints, or resulting limits are rejected.

## Changes

- `packages/doenetml-worker-javascript/src/components/Graph.js`: add shared inverse-definition logic for the `xscale` and `yscale` state variables.
- `packages/doenetml-worker-javascript/src/test/tagSpecific/graph.test.ts`: add tests covering zero/non-zero midpoints, difference matching the set scale, round-trip read-back, `identicalAxisScales`, `fixAxes` refusal, non-finite values, and graph-parent limit propagation.
- Changeset added.

## Testing

`npx vitest run --root packages/doenetml-worker-javascript src/test/tagSpecific/graph.test.ts` — all 36 tests pass.

🤖 Generated with [GitHub Copilot](https://github.com/features/copilot)

### CODE REVIEW

File: packages/doenetml-worker-javascript/src/components/Graph.js [hunk 1]
```diff
@@ -1828,6 +1828,48 @@ export default class Graph extends BlockComponent {
...
+                let desiredMax = midpoint + desiredScale / 2;
+
+                if (
+                    !Number.isFinite(desiredScale) ||
+                    !Number.isFinite(midpoint) ||
+                    !Number.isFinite(desiredMin) ||
+                    !Number.isFinite(desiredMax)
+                ) {
+                    return { success: false };
+                }
```
[the remark this answers is not in the document]
Comment (dqnykamp) [reply packages/doenetml-worker-javascript/src/components/Graph.js:1859]: Good catch — addressed in 8e8931fdd. The `xscale`/`yscale` inverse now rejects non-positive requested scales (in addition to the finiteness checks), since a non-positive scale would set `min === max` (or `min > max`) and break consumers that treat the scale as a positive magnitude. Added tests covering zero and negative requests.

🤖 Generated with [GitHub Copilot](https://github.com/features/copilot)
